server/src-lib/Hasura/RQL/Types/RemoteRelationship.hs

```diff
@@ -0,0 +1,337 @@
+{-# LANGUAGE DisambiguateRecordFields #-}
+{-# LANGUAGE NamedFieldPuns           #-}
+{-# LANGUAGE RecordWildCards          #-}
+
+module Hasura.RQL.Types.RemoteRelationship where
+
+import           Data.Aeson.Casing
+import           Data.Aeson.TH
+import           Hasura.Prelude
+import           Hasura.RQL.Instances          ()
+import           Hasura.RQL.Types.Common
+import           Hasura.SQL.Types
+
+import           Data.Aeson                    as A
+import qualified Data.Aeson.Types              as AT
+import           Data.HashMap.Strict           (HashMap)
+import qualified Data.HashMap.Strict           as HM
+import           Data.List.NonEmpty            (NonEmpty (..))
+import           Data.Scientific
+import           Data.Set                      (Set)
+
+import qualified Data.Text                     as T
+import           Data.Validation
+import qualified Database.PG.Query             as Q
+import           Instances.TH.Lift             ()
+import qualified Language.GraphQL.Draft.Syntax as G
+import           Language.Haskell.TH.Syntax    (Lift)
+
+import qualified Hasura.GraphQL.Validate.Types as VT
+import           Hasura.RQL.Types.RemoteSchema
+
+data RemoteField =
+  RemoteField
+    { rmfRemoteRelationship :: !RemoteRelationship
+    , rmfGType              :: !G.GType
+    , rmfParamMap           :: !(HashMap G.Name VT.InpValInfo)
+    }
+  deriving (Show, Eq, Lift)
+
+data RemoteRelationship =
```

Comment:
This definition can/should be changed to incorporate for a more holistic remote relationship i.e. (remote-to-remote), (remote-to-db)

Reply:
This can be changed as we add features right? Or are you concerned with extending the API while staying backwards compatible?

Reply:
Yep, keeping it backward compatible. Also, we already know what's going to come up.

Reply:
Do you think we can make a sum type called `data SourceFields = TableSource QualifiedTable (Set FieldName)` ?

Reply:
Honestly I don't think it's worth trying to do this right now. It seems likely it will need to change again once we actually get deep into implementation, right?

Reply:
I agree but we already have an implementation to refer: https://github.com/hasura/graphql-engine/blob/985e1cd6c58757ffcd3138572d573d83ec9cded1/server/src-lib/Hasura/RQL/Types/RemoteRelationship.hs#L341

I had to create a `RemoteToRemoteRelationship` type only because the types were so different. 

Do you think we should do something like:

```
data RemoteRelationshipNew =
  RemoteRelationshipNew
  { type: TableToRemote
  , configuration: RemoteRelationship
  }

This seems to affect much less. I just want to avoid some backward incompatibility nightmare later.